### PR TITLE
refactor: only pass --outDir flag when needed

### PR DESCRIPTION
**Pull request by alexeagle**
Reduces the severity of the workaround needed for TS #59036

### CODE REVIEW

**Comment by alexeagle** [comment]
Note, the failing test is red on `main` as well.

File: ts/private/ts_project.bzl [hunk 1]
```diff
@@ -125,24 +125,28 @@ See https://github.com/aspect-build/rules_ts/issues/361 for more details.
     # Add user specified arguments *before* rule supplied arguments
     common_args.extend(ctx.attr.args)
 
-    outdir = _lib.join(
-        ctx.label.workspace_root,
-        _lib.join(ctx.label.package, ctx.attr.out_dir) if ctx.attr.out_dir else ctx.label.package,
-    )
+    if (ctx.attr.out_dir and ctx.attr.out_dir != ".") or ctx.attr.root_dir:
+        # TODO: add validation that excludes is non-empty in this case, as passing the --outDir or --declarationDir flag
+        # to TypeScript causes it to set a default for excludes such that it won't find our sources that were copied-to-bin.
```
Comment (jbedard) [inline ts/private/ts_project.bzl:130]: Can we put a reference to https://github.com/microsoft/TypeScript/issues/59036 and the rules_js issue about it?

Comment (alexeagle) [reply ts/private/ts_project.bzl:130]: done

**Review by jbedard — APPROVED** [review]

**Comment by jbedard** [comment]
Double check the worker CI failure before any merging 🙏

**Comment by gzm0** [comment]
FWIW: This change causes the `outDir` config in `tsconfig.json` to be used while previously it wasn't.

Maybe it is worth adjusting the config validator to check it isn't there / it is consistent.

**Comment by alexeagle** [comment]
Yes jbedard already noticed that at some customers, I think there's an issue and a fix already.

**Pull request merged**